security/pkg/pki/ca/ca.go

```diff
@@ -150,15 +158,22 @@ func NewSelfSignedIstioCAOptions(ctx context.Context,
 	b := backoff.NewExponentialBackOff(backoff.DefaultOption())
 	err = b.RetryWithContext(ctx, func() error {
 		caCertName = CASecret
+		// This seems wrong: with in-cluster Istiod the /etc/cacerts is checked first, and if found it is used.
+		// This code path is called either when /etc/cacerts is not found (so the secret doesn't exist) or if we run Istiod outside of the cluster.
+
+		// We should always check the cacerts first.
```

Comment:
Could you clarify what you mean here? In `createIstioCA` we first check for the existence of a file mounted signing bundle (i.e. we always check the `cacerts` first). This code path pertains to istiod generating it's own self signed root certificate because a plugged in one wasn't detected. That is a separate code path.